timelock-api/src/main/java/com/palantir/atlasdb/timelock/api/GenericNamedMinTimestamp.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.api;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.SafeArg;
+import org.immutables.value.Value;
+
+@JsonDeserialize(as = ImmutableGenericNamedMinTimestamp.class)
+@JsonSerialize(as = ImmutableGenericNamedMinTimestamp.class)
+@Value.Immutable
+public interface GenericNamedMinTimestamp {
```

Comment:
I chose not to create a specific object for the immutable timestamp (i.e. a "ReservedTs..").

The API enforces the protection by virtue of this class's serde and the middleware will just take strings in. By the time we get there, we assume things are ok. And, when the immutable ts APIs are called, we will use the constant below